test/config-luatest/autoexpel_test.lua

```diff
@@ -0,0 +1,341 @@
+local yaml = require('yaml')
+local t = require('luatest')
+local treegen = require('luatest.treegen')
+local justrun = require('luatest.justrun')
+local cbuilder = require('luatest.cbuilder')
+local cluster = require('test.config-luatest.cluster')
+
+local g = t.group()
+
+g.before_all(cluster.init)
+g.after_each(cluster.drop)
+g.after_all(cluster.clean)
+
+-- {{{ Helpers
+
+local function assert_rs(server, exp)
+    server:exec(function(exp)
+        local res = box.space._cluster:pairs():map(function(t)
+            return t[3]
+        end):totable()
+        table.sort(res)
+        t.assert_equals(res, exp)
+    end, {exp})
+end
+
+local function reload(server)
+    server:exec(function()
+        local config = require('config')
+
+        config:reload()
+    end)
+end
+
+local function promote(server)
+    server:exec(function()
+        local fiber = require('fiber')
+
+        t.helpers.retrying({timeout = 60}, function()
+            -- Run box.ctl.promote() in background, because it
+            -- seems that there are situations, when this call
+            -- continues infinitely.
+            --
+            -- TODO: This behavior is surprising and we possibly
+            -- should consider it as a bug.
+            fiber.new(box.ctl.promote)
```

Comment:
I'll attempt to shrink the reproducer to some extent, will collect additional information and will file an issue against core team. Let's consider it as a separate activity, not a blocker for this patch.

Reply:
Filed #10836. Updated the test with the issue number.

```diff
diff --git a/test/config-luatest/autoexpel_test.lua b/test/config-luatest/autoexpel_test.lua
index 1485cd8475..2c5c8745bb 100644
--- a/test/config-luatest/autoexpel_test.lua
+++ b/test/config-luatest/autoexpel_test.lua
@@ -36,12 +36,9 @@ local function promote(server)
         local fiber = require('fiber')
 
         t.helpers.retrying({timeout = 60}, function()
-            -- Run box.ctl.promote() in background, because it
-            -- seems that there are situations, when this call
-            -- continues infinitely.
-            --
-            -- TODO: This behavior is surprising and we possibly
-            -- should consider it as a bug.
+            -- gh-10836: run box.ctl.promote() in background,
+            -- because it seems that there are situations, when
+            -- this call continues infinitely.
             fiber.new(box.ctl.promote)
             t.helpers.retrying({timeout = 1}, function()
                 assert(box.info.ro == false)
```